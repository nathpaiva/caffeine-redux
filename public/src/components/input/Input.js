import React from 'react';

const style = {
  styleInputDisabled: {
    border: '1px solid #CCCCCC',
    padding: '0 5px',
    margin: '2px 0',
    width: '100%',
    height: '32px',
    opacity: '.5'
  },
  styleInput: {
    border: '1px solid #CCCCCC',
    padding: '0 5px',
    margin: '2px 0',
    width: '100%',
    borderRadius: '2px',
    height: '32px',
  },
  styleLabel: {
    fontSice: '12px',
    fontFamily: '"Roboto", sans-serif'
  }
}

const Input = ({ type, id, text, value, label, disabled, inputRef, defaultValue }) => {
  return (
    <div>
      {label ? <label style={style.styleLabel} htmlFor={id}>{label}</label> : ''}
      <input ref={inputRef}
        style={!disabled ? style.styleInput : style.styleInputDisabled}
        defaultValue={defaultValue ? defaultValue : ''}
        disabled={disabled ? disabled : false}
        type={type}
        id={id} name={id} placeholder={text} />
    </div>
  )
};

export default Input;
