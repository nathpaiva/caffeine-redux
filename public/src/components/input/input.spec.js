'use stict';

import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';


describe('#Test Input Component', () => {
  it('#Should Input to mach snapshot without label', () => {
    const tree = renderer.create(<Input type='email' id='email' text='email' value='nath@nath.com.br' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('#Should Input to mach snapshot with label', () => {
    const tree = renderer.create(<Input type='email' id='email' text='email' value='nath@nath.com.br' label='Email' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
