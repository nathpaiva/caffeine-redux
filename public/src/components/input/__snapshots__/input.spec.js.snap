// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test Input Component #Should Input to mach snapshot with label 1`] = `
<div>
  <label
    htmlFor="email"
    style={
      Object {
        "fontFamily": "\\"Roboto\\", sans-serif",
        "fontSice": "12px",
      }
    }
  >
    Email
  </label>
  <input
    defaultValue=""
    disabled={false}
    id="email"
    name="email"
    placeholder="email"
    style={
      Object {
        "border": "1px solid #CCCCCC",
        "borderRadius": "2px",
        "height": "32px",
        "margin": "2px 0",
        "padding": "0 5px",
        "width": "100%",
      }
    }
    type="email"
  />
</div>
`;

exports[`#Test Input Component #Should Input to mach snapshot without label 1`] = `
<div>
  
  <input
    defaultValue=""
    disabled={false}
    id="email"
    name="email"
    placeholder="email"
    style={
      Object {
        "border": "1px solid #CCCCCC",
        "borderRadius": "2px",
        "height": "32px",
        "margin": "2px 0",
        "padding": "0 5px",
        "width": "100%",
      }
    }
    type="email"
  />
</div>
`;
